sentry-core/src/clientoptions.rs

```diff
@@ -11,6 +11,31 @@ use crate::{Integration, IntoDsn, TransportFactory};
 /// Type alias for before event/breadcrumb handlers.
 pub type BeforeCallback<T> = Arc<dyn Fn(T) -> Option<T> + Send + Sync>;
 
+/// The Session Mode of the SDK.
+///
+/// Depending on the use-case, the SDK can be set to two different session modes:
+///
+/// * **Application Mode Sessions**:
+///   This mode should be used for user-attended programs, which typically have
+///   a single long running session.
```

Comment:
_nit_: You could mention that the session spans the "application's" lifetime, which should make it more apparent why this is called "Application Mode".